### PR TITLE
Group bot accounts UI

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI for group bot accounts (listing, removing, adding).

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

image

image

image

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com` or any other user
2. Create a group
3. From the sidebar on the project show page, click `Settings` and then `Bot Accounts`
4. You should not see any bot accounts
5. Click the `New bot account` button
6. Fill in the token name, access level, option expiration date for token, and select at least 1 scope
7. Click `Submit`
8. You should see a table with the bot account that was created and above the table a section which lists the personal access token
9. Go to the `Members` page for the group
10. You should see the bot account that was created listed as a `Bot`
11. Go back to the `Bot Accounts` page for the group
12. Click the ellipsis for the bot account and click `Remove`. Confirm that you would like to remove the bot account
13. The table should update and no bot accounts should be listed
14. Go back to the `Members` page for the group
15. Verify that the bot account has also been removed from the group members

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by joshsadam — DISMISSED** [review]
This looks good to me.  It will be good once these are made into components during the refactor stage since they are used across groups and projects.

File: app/services/bots/create_service.rb [hunk 1]
```diff
@@ -10,11 +10,12 @@ def initialize(user = nil, namespace = nil, bot_type = nil, params = {}) # ruboc
       super(user, params)
       @namespace = namespace
 
+      current_count = @namespace.namespace_bots.count
       user_params = {
-        email: "#{namespace.type.downcase}_#{namespace.puid.downcase}_bot_#{SecureRandom.hex(5)}@iridanext.com",
+        email: "#{namespace.puid}_BOT_#{format('%03d', (current_count + 1))}@iridanext.com",
```
Comment (ericenns) [inline app/services/bots/create_service.rb:15]: **note:** I guess rails is downcasing the emails, so I guess it wasn't as important to uppercase it here.

File: app/services/bots/create_service.rb [hunk 2]
```diff
@@ -10,11 +10,12 @@ def initialize(user = nil, namespace = nil, bot_type = nil, params = {}) # ruboc
...
         user_type: bot_type,
-        first_name: namespace.type.downcase,
-        last_name: 'Bot'
+        first_name: namespace.puid,
+        last_name: "BOT #{format('%03d', (current_count + 1))}"
```
Comment (ericenns) [inline app/services/bots/create_service.rb:18]: **nitpick (blocking):** Can we change this to `Bot 001`, I don't think `BOT` needs to be all upper case in this context.

File: app/views/layouts/groups.html.erb [hunk 3]
```diff
@@ -31,12 +31,16 @@
               selected: @current_page == t(:"groups.sidebar.history").downcase
             ) %>
           <% end %>
-          <% if allowed_to?(:update?, @group) %>
-            <%= render section.with_item(
-              url: edit_group_path(@group),
-              icon: "cog_6_tooth",
-              label: t(:"groups.sidebar.settings"),
-              selected: @current_page == t(:"groups.sidebar.settings").downcase
+          <%= render section.with_multi_level_menu(title: t(:"groups.sidebar.settings"), icon: "cog_6_tooth") do |multi_level_menu| %>
```
Comment (ericenns) [inline app/views/layouts/groups.html.erb:34]: **nitpick (blocking):** I think this whole section should be surrounded with `allowed_to?(:update?, @group)` as non `Maintainer` or `Owner' should not see this section.

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -156,9 +156,9 @@ user30:
   first_name: User
   last_name: "30"
 
-<% 21.times do |n| %>
+<% 22.times do |n| %>
 user_group_bot_account<%= (n) %>:
-  email: <%= "group_INXT_GRP_AAAAAAAAAA_bot_A3d#{n}Xc@localhost".downcase %>
+  email: <%= "group_INXT_GRP_AAAAAAAAAA_bot_A5d#{n}Xc@localhost".downcase %>
```
Comment (ericenns) [inline test/fixtures/users.yml:161]: **nitpick (blocking):** Can you update this to the new format that we talked about.

**Review by ericenns — CHANGES_REQUESTED** [review]
Just a few changes requested.

File: app/services/bots/create_service.rb (hunk 1 above)
Comment (deepsidhu85) [reply app/services/bots/create_service.rb]: Yeah rails automatically does that. Updated in d55982d7e963f7e8365a2f47f03f3b8060ec9230

File: app/services/bots/create_service.rb (hunk 2 above)
Comment (deepsidhu85) [reply app/services/bots/create_service.rb]: Updated in d55982d7e963f7e8365a2f47f03f3b8060ec9230

File: test/fixtures/users.yml (hunk 4 above)
Comment (deepsidhu85) [reply test/fixtures/users.yml]: Updated in d55982d7e963f7e8365a2f47f03f3b8060ec9230

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**